### ISSUE TITLE
Getting Started guide causes ESLint complaints in .eslintrc.(c)js

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6825]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

When starting a TypeScript repository from scratch using our docs, we tell folks to use this as their `.eslintrc.cjs` config:

```js
module.exports = {
  extends: ['eslint:recommended', 'plugin:@typescript-eslint/recommended'],
  parser: '@typescript-eslint/parser',
  plugins: ['@typescript-eslint'],
  root: true,
};
```

However, that results in an ESLint error on the `.eslintrc.cjs` file when viewed in VS Code!

```plaintext
'module' is not defined. eslint [no-undef](https://eslint.org/docs/rules/no-undef)
```

![Screenshot of the above lint error in VS Code](https://user-images.githubusercontent.com/3335181/229380709-867a0369-bb92-42bb-852e-801be0413542.png)

This is confusing. We should fix the code and/or docs so that our first run experience doesn't include ESLint errors for problems that don't exist. I'm not sure what the correct solution is, though. A few options:

* Use [`env: { node: true }`](https://eslint.org/docs/latest/use/configure/language-options#specifying-environments) for specifically this file or files like it?
* Explicitly disable the rule
* Don't change the config, and add a warning to the docs

...I don't love any of those 🤔. How are users "supposed" to fix this?

_(using the [flat config system](https://eslint.org/docs/latest/use/configure/configuration-files-new) is not a viable solution yet, as flat configs are still considered experimental)_

This issue came out of a Twitch stream with @Jutanium. 👋

### Affected URL(s)

https://typescript-eslint.io/getting-started

**Comment by armano2** [comment]
to be precise you actuallyw ant to enable `/* eslint-env commonjs */` not `/* eslint-env node */`, unless you are working in actual node env

this is described in https://eslint.org/docs/latest/use/configure/language-options

**Comment by JoshuaKGoldberg** [comment]
Related: https://github.com/eslint/create-config/issues/59
